torch/linalg/__init__.py

```diff
@@ -348,16 +352,16 @@
              In this implmentation, the returned eigenvectors are normalized to have norm
              `1` and largest real component.
 
-.. warning:: This function assumes that :attr:`A` is `diagonalizable`_ (e.g. when all the
-             eigenvalues are different). If it is not diagonalizable, the returned
-             eigenvalues will be correct but :math:`A \neq V \operatorname{diag}(\Lambda)V^{-1}`.
-
-.. warning:: This function does not yet support autograd.
+.. warning:: Gradients computed using `V` will only be finite when :attr:`A` does not have repeated eigenvalues.
+             Furthermore, if the distance between any two eigenvalues is close to zero,
+             the gradient will be numerically unstable, as it depends on the eigenvalues
+             :math:`\lambda_i` through the computation of
+             :math:`\frac{1}{\min_{i \neq j} \lambda_i - \lambda_j}`.
 
 .. seealso::
 
         :func:`torch.linalg.eigvals` computes only the eigenvalues.
-        However, that function is not differentiable.
+        Unlike :func:`torch.linalg.eig`, the gradients of this function are always numerically stable.
```

Comment:
The very last sentence - always stable, yet at the very top there is a condition that makes things unstable, makes for a confusion.

Reply:
I am not sure I understand this comment.

The `warning` section starts saying that "the gradients are unstable when `V` is used", which sort of implies this comment (when we do not use `V`, for example, in `eigvals`, we do not have numerical issues)

Do you have a proposal

Reply:
I suggesting writing :func:\`torch.linalg.eivgals\` explicitly instead of "this," which is unclear what it refers to  if not reading carefully